tests/stress/ebpf_stress_tests.cpp

```diff
@@ -0,0 +1,475 @@
+
+        REQUIRE(bpf_map_lookup_elem(dropped_packet_map_fd, &key, &value) == EBPF_SUCCESS);
+        REQUIRE(value == 1000);
+
+        // Do some more basic validations.
+        REQUIRE(bpf_map_delete_elem(dropped_packet_map_fd, &key) == EBPF_SUCCESS);
+
+        REQUIRE(bpf_map_lookup_elem(dropped_packet_map_fd, &key, &value) == EBPF_SUCCESS);
+        REQUIRE(value == 0);
+
+        // Detach link.
+        hook.detach_link(link);
+        hook.close_link(link);
+
+        bpf_object__close(object);
+    }
+
+    LOG_INFO("{} done. Iterations: {}", test_params.file_name.c_str(), count);
+}
+
+static void
+_printk_stress_thread_func(const stress_test_thread_context& test_params)
+{
+    uint32_t count{0};
+    using sc = std::chrono::steady_clock;
+    auto endtime = sc::now() + std::chrono::seconds(test_params.runtime);
+    while (sc::now() < endtime) {
+
+        single_instance_hook_t hook(EBPF_PROGRAM_TYPE_BIND, EBPF_ATTACH_TYPE_BIND);
+
+        LOG_VERBOSE(
+            "{}({}): Instantiating program_helper. Iteration #: {}", __func__, test_params.thread_index, count++);
+
+        uint32_t ifindex = test_params.thread_index;
+        program_load_attach_helper_t program_helper(
+            test_params.file_name.c_str(),
+            test_params.prog_type,
+            "func",
+            test_params.execution_type,
+            &ifindex,
+            sizeof(ifindex),
+            hook);
+
+        REQUIRE(program_helper.get_object() != nullptr);
+    }
+
+    LOG_INFO("{} done. Iterations: {}", test_params.file_name.c_str(), count);
+}
+
+// These objects should be created just _once_ per (test) process. This is only needed for user mode tests that use the
+// user mode 'mock' framework.  Note that these cannot be created globally and _must_ be created in the context of a
+// a Catch2 test 'session' (Per Catch2 documentation, Catch2's exception framework is apparently not quite ready until
+// then).  This is an issue in our usage as we make extensive use of Catch2's REQUIRE verification/validation macros
+// (based on this framework) during the creation of these objects;
+static _test_helper_end_to_end* _test_helper{nullptr};
+static program_info_provider_t* _bind_program_info{nullptr};
+static program_info_provider_t* _xdp_program_info{nullptr};
+
+static std::once_flag _um_test_init_done;
+static void
+um_test_init()
+{
+    std::call_once(_um_test_init_done, [&]() {
+        _test_helper = new _test_helper_end_to_end;
+        REQUIRE(_test_helper != nullptr);
+
+        _bind_program_info = new program_info_provider_t(EBPF_PROGRAM_TYPE_BIND);
+        REQUIRE(_bind_program_info != nullptr);
+
+        _xdp_program_info = new program_info_provider_t(EBPF_PROGRAM_TYPE_XDP);
+        REQUIRE(_xdp_program_info != nullptr);
+
+        LOG_INFO("One time initialization complete");
+    });
+}
+
+using test_thread_func_t = void (*)(const stress_test_thread_context& test_params);
```

Comment:
Coding style guidance says to avoid abbreviations.  Can you expand "func" to "function" throughout?  Or is there some C++ API this is trying to match?

Reply:
Fixed.